spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherProperties.java

```diff
@@ -43,4 +71,16 @@ public void setModules(String[] modules) {
 		return modules;
 	}
 
+	@PostConstruct
+	public void deduceModuleArguments() throws Exception {
```

Comment:
Didn't you say order was important? I assumed this was for consumer _then_ producer reasons. Note that the launcher, even it its current form, just iterates on that array. Or has it been reversed elsewhere already?